text/2021-04-23-builtin-gc-worker.md

```diff
@@ -12,26 +12,29 @@ GC worker is an important component for TiKV that deletes outdated MVCC data so
 
 According to the [documentation](https://docs.pingcap.com/tidb/stable/garbage-collection-overview) for the **current** GC worker, the GC worker processes the following steps in a GC cycle:
 
-1. Calculate the minimal timestamp among all living SQL sessions, earlier than which are safe to GC, called `Service Safepoint`.
-2. [Upload](https://github.com/pingcap/kvproto/blob/8ecb5e46d7f5f7952a1a8d262b54f61dc8de1ef3/proto/pdpb.proto#L73) the `Service Safepoint` to PD, and get the minimal `Service Safepoint` among all Tools (e.g. CDC) from the response.
-3. Resolve all locks whose timestamp is earlier than the `Minimal Service Safepoint`.
-4. [Push](https://github.com/pingcap/kvproto/blob/8ecb5e46d7f5f7952a1a8d262b54f61dc8de1ef3/proto/pdpb.proto#L71) the `Minimal Service Safepoint` as `GCSafepoint` to PD.
+1. Calculate the minimal timestamp among all living SQL sessions, earlier than which are safe to GC, called `Session Safepoint`.
+2. [Fetch](https://github.com/pingcap/kvproto/blob/8ecb5e46d7f5f7952a1a8d262b54f61dc8de1ef3/proto/pdpb.proto#L73) the `Service Safepoint` from PD, which is the minimal snapshot timestamp requirement among all tools (e.g. CDC, BR).
```

Comment:
Actually it's not only *fetching*. TiDB it self also registers a service safepoing with id = `"gc_worker"`.

Reply:
edited